### ISSUE TITLE
apps/cufflinks/2.2.2.20150701 install error

### ISSUE THREAD

**Issue by vlj91** [alces-software/packager-base#158]
```
Preparing to install main/apps/cufflinks/2.2.2.20150701

WARNING: Package requires the installation of the following:
  main/libs/eigen/3.2.4, main/libs/boost/1.55.0

Install these dependencies first?

Proceed (Y/N)? y
Preparing to install main/libs/eigen/3.2.4
Installing main/libs/eigen/3.2.4
Importing libs-eigen-3.2.4-el7.tar.gz

 > Fetching archive
        Download ... OK

 > Preparing import
         Extract ... OK
          Verify ... OK

 > Processing libs/eigen/3.2.4/gcc-4.8.5
       Preparing ... OK
       Importing ... OK
     Permissions ... OK

 > Finalizing import
          Update ... OK
    Dependencies ... OK

Installation complete.
Preparing to install main/libs/boost/1.55.0
Installing main/libs/boost/1.55.0
Importing libs-boost-1.55.0-el7.tar.gz

 > Fetching archive
        Download ... OK

 > Preparing import
         Extract ... OK
          Verify ... OK

 > Processing libs/boost/1.55.0/gcc-4.8.5+openmpi-1.8.5+python-2.7.8
       Preparing ... OK
       Importing ... OK
     Permissions ... OK

 > Finalizing import
          Update ... OK
    Dependencies ... OK

Installation complete.
Installing main/apps/cufflinks/2.2.2.20150701
Importing apps-cufflinks-2.2.2.20150701-el7.tar.gz

 > Fetching archive
        Download ... OK

 > Preparing import
         Extract ... OK
          Verify ... OK

 > Processing apps/cufflinks/2.2.2.20150701/gcc-4.8.5+boost-1.55.0+samtools-0.1.19+eigen-3.2.4
       Preparing ... NOTICE: importing requirements
--------------------------------------------------------------------------------
error: undefined method `metadata' for nil:NilClass. Use --trace to view backtrace
[root@login1(vlj) ~]# al gr inst -b main/apps/cufflinks/2.2.2.20150701 --trace
Preparing to install main/apps/cufflinks/2.2.2.20150701
Installing main/apps/cufflinks/2.2.2.20150701
Importing apps-cufflinks-2.2.2.20150701-el7.tar.gz

 > Fetching archive
        Download ... SKIP (Existing source file detected)

 > Preparing import
         Extract ... OK
          Verify ... OK

 > Processing apps/cufflinks/2.2.2.20150701/gcc-4.8.5+boost-1.55.0+samtools-0.1.19+eigen-3.2.4
       Preparing ... NOTICE: importing requirements
--------------------------------------------------------------------------------
/opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:162:in `block (3 levels) in import_package': undefined method `metadata' for nil:NilClass (NoMethodError)
        from /opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:153:in `each'
        from /opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:153:in `block (2 levels) in import_package'
        from /opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:137:in `catch'
        from /opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:137:in `block in import_package'
        from /opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:136:in `each'
        from /opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:136:in `import_package'
        from /opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:95:in `block in import'
        from /opt/clusterware/opt/ruby/lib/ruby/2.2.0/tmpdir.rb:88:in `mktmpdir'
        from /opt/clusterware/lib/ruby/lib/alces/packager/archive_importer.rb:73:in `import'
        from /opt/clusterware/lib/ruby/lib/alces/packager/definition_handler.rb:119:in `install_defn'
        from /opt/clusterware/lib/ruby/lib/alces/packager/definition_handler.rb:58:in `install'
        from /opt/clusterware/lib/ruby/lib/alces/packager/definition_handler.rb:31:in `install'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:183:in `block (2 levels) in install'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:216:in `call'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:216:in `with_resource'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:198:in `with_definition'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:182:in `block in install'
        from /opt/clusterware/lib/ruby/vendor/ruby/2.2.0/gems/dm-core-1.2.1/lib/dm-core.rb:263:in `block in repository'
        from /opt/clusterware/lib/ruby/vendor/ruby/2.2.0/gems/dm-core-1.2.1/lib/dm-core/repository.rb:114:in `scope'
        from /opt/clusterware/lib/ruby/vendor/ruby/2.2.0/gems/dm-core-1.2.1/lib/dm-core.rb:263:in `repository'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:223:in `with_depot'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:181:in `install'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:50:in `block in method_missing'
        from /opt/clusterware/opt/ruby/lib/ruby/gems/2.2.0/gems/bundler-1.10.6/lib/bundler.rb:245:in `block in with_clean_env'
        from /opt/clusterware/opt/ruby/lib/ruby/gems/2.2.0/gems/bundler-1.10.6/lib/bundler.rb:232:in `with_original_env'
        from /opt/clusterware/opt/ruby/lib/ruby/gems/2.2.0/gems/bundler-1.10.6/lib/bundler.rb:238:in `with_clean_env'
        from /opt/clusterware/lib/ruby/lib/alces/packager/handler.rb:49:in `method_missing'
        from /opt/clusterware/lib/ruby/vendor/ruby/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:179:in `call'
        from /opt/clusterware/lib/ruby/vendor/ruby/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
        from /opt/clusterware/lib/ruby/vendor/ruby/2.2.0/gems/commander-4.3.5/lib/commander/runner.rb:426:in `run_active_command'
        from /opt/clusterware/lib/ruby/vendor/ruby/2.2.0/gems/commander-4.3.5/lib/commander/runner.rb:68:in `run!'
        from /opt/clusterware/lib/ruby/vendor/ruby/2.2.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
        from /opt/clusterware/libexec/actions/gridware:75:in `<main>'
```

**Comment by mjtko** [comment]
This looks like a code path that's not properly erroring out when trying to process a missing requirement.

**Comment by mjtko** [comment]
Odd: for some reason the metadata held in the binary build of cufflinks has mandated boost `1.49.0`:

```
  :requirements:
  - libs/boost = 1.49.0
```

Not quite sure where that's come from yet.

**Comment by mjtko** [comment]
Ah, this has been caused by alces-software/clusterware#93.  Will need to rebuild this binary when CI is updated to Clusterware 1.5.0.

**Comment by mjtko** [comment]
Closed via #182.
